aea/helpers/dialogue/base.py

```diff
@@ -736,20 +740,13 @@ def __init__(
         self._dialogue_nonce = 0
         self._dialogue_stats = DialogueStats(end_states)
 
-        if message_class is not None:
-            assert issubclass(message_class, Message)
+        assert issubclass(message_class, Message)
         self._message_class = message_class
 
-        if dialogue_class is not None:
-            assert issubclass(dialogue_class, Dialogue)
+        assert issubclass(dialogue_class, Dialogue)
         self._dialogue_class = dialogue_class
 
-        if role_from_first_message is not None:
-            self._role_from_first_message = role_from_first_message
-        else:
-            self._role_from_first_message = (
-                self.role_from_first_message
-            )  # pragma: no cover
+        self._role_from_first_message = role_from_first_message
```

Comment:
add check on role_from_first_message

Reply:
I did implement some checks using `inspect.signature` but that might be a bit of an overkill IMO, wouldn't the type hint (`Callable...`) be sufficient?

Reply:
No; because type hints are not used at runtime.